articles/azure/azs-how-configure-custom-script-extension.md

```diff
@@ -0,0 +1,338 @@
+| \$CommandToExecute        | The command to execute                          | <form oninput="result.value=commandtoexecute.value;result2.value=commandtoexecute.value;result3.value=commandtoexecute.value;result4.value=commandtoexecute.value;result5.value=commandtoexecute.value;result6.value=commandtoexecute.value;result7.value=commandtoexecute.value;result8.value=commandtoexecute.value" id="commandtoexecute" style="display: inline;"><input type="text" id="commandtoexecute" name="commandtoexecute" style="display: inline;" placeholder="sh SetRootPassword.sh Password123!"/></form> |
+
+Select the desired deployment option:
+
+## [New VM](#tab/tabid-1)
+
+See documentation here on [how to create a new VM on Azure Stack](azs-how-create-vm-ps.md).
+
+## [Existing VM](#tab/tabid-2)
+
+***
+
+### [Linux](#tab/tabid-a/tabid-1)
+
+Add the following code to the end of the VM creation script when creating a Linux VM on Azure Stack.
+
+#### Local disk
+
+<pre><code class="language-PowerShell">$CustomScriptFileName = "<output form="customscriptfilename" name="result" style="display: inline;">SetRootPassword.sh</output>"
+$ContainerName = "<output form="containername" name="result" style="display: inline;">customscriptextension</output>"
+$CustomScriptStorageAccountName = "<output form="customscriptstorageaccountname" name="result" style="display: inline;">customscript<span id="RandNum"></span></output>".ToLower()
+$FilePath = "<output form="filepath" name="result" style="display: inline;">C:\Users\User1\SetRootPassword.sh</output>"
+$CommandToExecute = "<output form="commandtoexecute" name="result" style="display: inline;">sh SetRootPassword.sh Password123!</output>"
+
+# Create a new storage account
+Write-Output -InputObject "Creating storage account and container"
+$StorageAccount = New-AzureRmStorageAccount -Location $Location -ResourceGroupName $RGName -Type "Standard_LRS" -Name $CustomScriptStorageAccountName
+
+# Get storage account context
+$Context = $StorageAccount.Context
+$Container = New-AzureStorageContainer -Name $ContainerName -Context $Context
+
+# Retrieve storage account key
+$StorageAccountKey = (Get-AzureRmStorageAccountKey -ResourceGroupName $RGName -Name $CustomScriptStorageAccountName).Value[0]
+
+# Retrieve storage blob endpoint
+$ScriptBlobUrl = $Container.Context.BlobEndPoint
+
+# Upload script extension to the storage account
+Write-Output -InputObject "Uploading custom script extension to storage account"
+Set-AzureStorageBlobContent -File $FilePath -Container $ContainerName -Blob $CustomScriptFileName -Context $Context
+
+# Creating script location string
+$ScriptLocation = $ScriptBlobUrl + "$ContainerName/" + $CustomScriptFileName
+
+# Add custom script extension to Linux VM
+Write-Output -InputObject "Adding custom script extension to VM"
+$Extensions = Get-AzureRmVMExtensionImage -Location "frn00006" -PublisherName "Microsoft.Azure.Extensions" -Type "CustomScript"
```

Comment:
Can we pull the location either directly from the stack or from a variable in the creation script?